flexget/components/managed_lists/lists/pending_list/api.py

```diff
@@ -264,6 +278,41 @@ def post(self, list_id, session=None):
         return response
 
 
+@pending_list_api.route('/<int:list_id>/entries/batch')
+@api.doc(params={'list_id': 'ID of the list'})
+@api.response(NotFoundError)
+class PendingListEntriesBatchAPI(APIResource):
+    @api.response(201, model=pending_lists_entries_return_schema)
+    @api.response(204)
+    @api.validate(model=pending_list_batch_schema)
+    @api.doc(description='Approve, reject, or remove multiple entries')
+    def post(self, list_id, session=None):
+        """Perform operations on multiple entries"""
+        data = request.json
+        entry_ids = data.get('ids')
+        operation = data.get('operation')
+        try:
+            entries = db.get_entries_by_list_id(list_id, entry_ids=entry_ids, session=session)
+        except NoResultFound:
+            raise NotFoundError('could not find entries in list %d' % list_id)
+
+        response = None
+        if operation == 'remove':
+            for entry in entries:
+                session.delete(entry)
+            response = jsonify([])
+            response.status_code = 204
+        else:
+            approved = operation == 'approve'
+            for entry in entries:
+                entry.approved = approved
+            response = jsonify([entry.to_dict() for entry in entries])
```

Comment:
i think jsonify can take the generator itself, meaning the `[]` is (probably) redundant

Reply:
I tried , it errored (TypeError: Object of type generator is not JSON serializable)